### PR TITLE
add detection for cyclic loops inside styled output feature

### PR DESCRIPTION
**Pull request by MikeVanMourik**
asana: https://app.asana.com/0/1201027711166952/1205169244716095/f
Add cyclic loop detection to styled output

### CODE REVIEW

File: Api/Modules/Queries/Services/StyledOutputService.cs [hunk 1]
```diff
@@ -9,6 +9,7 @@
 using Api.Core.Services;
 using Api.Modules.Customers.Interfaces;
 using Api.Modules.Queries.Interfaces;
+using DocumentFormat.OpenXml.Office2010.Excel;
```
Comment (GilianJuice) [inline Api/Modules/Queries/Services/StyledOutputService.cs:12]: Please remove this using. Your code has nothing to do with Excel as far as I know?

File: Api/Modules/Queries/Services/StyledOutputService.cs [hunk 2]
```diff
@@ -51,8 +52,21 @@ public class StyledOutputService : IStyledOutputService, IScopedService
         }
 
         /// <inheritdoc />
-        public async Task<ServiceResult<JToken>> GetStyledOutputResultJsonAsync(ClaimsIdentity identity, int id, List<KeyValuePair<string, object>> parameters, bool stripNewlinesAndTabs)
+        public async Task<ServiceResult<JToken>> GetStyledOutputResultJsonAsync(ClaimsIdentity identity, int id, List<KeyValuePair<string, object>> parameters, bool stripNewlinesAndTabs, List<int> inUseStyleIds = null)
         {
+            var usedIds = inUseStyleIds == null ? new List<int>() : new List<int>(inUseStyleIds);
```
Comment (GilianJuice) [inline Api/Modules/Queries/Services/StyledOutputService.cs:57]: This line can be made much shorter like this:
```c#
inUseStyleIds ??= new List<int>();
```
Why do you create a new variable here anyway and what is the `new List<int>(inUseStyleIds)` for?

**Review by GilianJuice — CHANGES_REQUESTED** [review]

File: Api/Modules/Queries/Services/StyledOutputService.cs (hunk 1 above)
Comment (MikeVanMourik) [reply Api/Modules/Queries/Services/StyledOutputService.cs]: you are right, not sure why that got added.. ill remove it

File: Api/Modules/Queries/Services/StyledOutputService.cs (hunk 2 above)
Comment (MikeVanMourik) [reply Api/Modules/Queries/Services/StyledOutputService.cs:56]: I need that copy so it resets

**Review by GilianJuice — APPROVED** [review]

**Pull request merged**